core/vault-worker.go

```diff
@@ -11,78 +11,233 @@
+		SecretThreshold: config.SecretService.VaultSecretThreshold,
 	}
-	r, err := c.SealStatus()
+
+	lc.Info(fmt.Sprintf("Vault Init Strategy (SSS parameters): Shares=%d Threshold=%d", initRequest.SecretShares, initRequest.SecretThreshold))
+
+	// Build Vault API full URL
+	url, err := url.Parse(config.SecretService.Scheme + "://" + config.SecretService.Server + ":" + config.SecretService.Port + vaultInitAPI)
+	// Build Vault HTTP/POST Request
+	jsonBlock, err := json.Marshal(&initRequest) //jsonBlock := []byte(`{}`)
 	if err != nil {
-		lc.Error(err.Error())
-		return true, err
+		lc.Error(fmt.Sprintf("Failed to build the Vault init request (SSS parameters): %s", err.Error()))
+		return 0, err
 	}
-	if r.Sealed == false {
-		lc.Info("Vault is in unseal status, nothing to do.")
-		return false, err
+	req, err := http.NewRequest(http.MethodPost, url.String(), bytes.NewBuffer(jsonBlock))
+
+	// Prepare the Header
+	req.Header.Set("Content-Type", contentType)
+
+	// POST the request
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		lc.Error(fmt.Sprintf("Failed to send the Vault init request: %s", err.Error()))
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	// Init request OK/KO ?
+	if resp.StatusCode != 200 {
+		lc.Error(fmt.Sprintf("Vault init request failed with status code: %s", resp.Status))
+		return resp.StatusCode, err
 	}
-	resp, err := c.Unseal(token)
+
+	// Grab the Vault init request response from HTTP body
+	initRequestResponseBody, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		fmt.Println(err.Error())
-		return true, err
+		lc.Error(fmt.Sprintf("Failed to fetch the Vault init request response body: %s", err.Error()))
+		return 0, err
+	}
+
+	// Build a JSON structure from the init request response HTTP body
+	var initResponse InitResponse
+	if err = json.Unmarshal(initRequestResponseBody, &initResponse); err != nil {
+		lc.Error(fmt.Sprintf("Failed to build the JSON structure from the init request response body: %s", err.Error()))
+		return 0, err
 	}
-	return resp.Sealed, err
+
+	if debug {
+		lc.Info(fmt.Sprintf("Vault Init Response: %s", initRequestResponseBody))
+	}
+
+	// Save the JSON structure to a file system JSON file
+	err = ioutil.WriteFile(config.SecretService.TokenFolderPath+"/"+config.SecretService.VaultInitParm, initRequestResponseBody, 0600)
+	if err != nil {
+		lc.Error(fmt.Sprintf("Fatal error creating Vault init response %s file, HTTP status: %s", config.SecretService.TokenFolderPath+"/"+config.SecretService.VaultInitParm, err.Error()))
+		return 0, err
+	}
+
+	lc.Info("Vault Initialization complete.")
+
+	return resp.StatusCode, nil
 }
 
-func checkProxyCerts(config *tomlConfig, secretBaseURL string, c *http.Client) (bool, error) {
-	cert, key, err := getCertKeyPair(config, secretBaseURL, c)
+// ----------------------------------------------------------
+func vaultUnseal(config *tomlConfig, httpClient *http.Client, debug bool) (sCode int, err error) {
```

Comment:
same for this function, I am not reviewing the logic here for consistency with the Vault API's